### PR TITLE
Fix flaky concurrent update on 6x

### PR DESCRIPTION
**Pull request by z-wenlin**
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/isolation2/sql/gdd/concurrent_update.sql [hunk 1]
```diff
@@ -28,20 +28,20 @@ DROP TABLE t_concurrent_update;
 2: UPDATE t_concurrent_update SET b=b+10 WHERE a=1;
 3: BEGIN;
 3: SET optimizer=off;
--- transaction 3 will wait transaction 1 on the segment
+-- transaction 3 will wait transaction 2 on the segment
 3&: UPDATE t_concurrent_update SET b=b+10 WHERE a=1;
 
 -- transaction 2 suspend before commit, but it will wake up transaction 3 on segment
-2: select gp_inject_fault('before_xact_end_procarray', 'suspend', dbid) FROM gp_segment_configuration WHERE role='p' AND content=-1;
+2: select gp_inject_fault('before_xact_end_procarray', 'suspend', '', 'isolation2test', '', 1, 1, 0, dbid) FROM gp_segment_configuration WHERE role='p' AND content=-1;
```
Comment (Aegeaner) [inline src/test/isolation2/sql/gdd/concurrent_update.sql:35]: I did not understand here, is this change to deduplicate multiple faults?

Comment (geekbosszhang) [reply src/test/isolation2/sql/gdd/concurrent_update.sql:35]: Add the database name here is to avoid other processes triggering the fault.

**Review by Aegeaner — APPROVED** [review]

File: src/backend/storage/ipc/procarray.c [hunk 2]
```diff
@@ -71,6 +71,7 @@
 #include "cdb/cdbvars.h"
 #include "utils/faultinjector.h"
 #include "utils/sharedsnapshot.h"
+#include "libpq/libpq-be.h"
```
Comment (ashwinstar) [inline src/backend/storage/ipc/procarray.c:74]: why do we need this include?

**Review by ashwinstar — APPROVED** [review]

File: src/backend/storage/ipc/procarray.c (hunk 2 above)
Comment (geekbosszhang) [reply src/backend/storage/ipc/procarray.c]: Removed.

**Pull request merged**